File: lib/src/main/java/com/nytimes/android/external/registerlib/GoogleServiceProviderTesting.java
package com.nytimes.android.external.registerlib;

import android.app.Activity;
import android.content.Context;
import android.support.annotation.NonNull;

import com.android.billingclient.api.BillingClientStateListener;
import com.android.billingclient.api.BillingClientTesting;
import com.android.billingclient.api.BillingFlowParams;
import com.android.billingclient.api.ConsumeResponseListener;
import com.android.billingclient.api.Purchase;
import com.android.billingclient.api.PurchaseHistoryResponseListener;
import com.android.billingclient.api.PurchasesUpdatedListener;
import com.android.billingclient.api.SkuDetailsParams;
import com.android.billingclient.api.SkuDetailsResponseListener;

public class GoogleServiceProviderTesting extends GoogleServiceProvider {

    private final BillingClientTesting billingClient;

    GoogleServiceProviderTesting(Context context, PurchasesUpdatedListener listener) {
        billingClient = new BillingClientTesting(context, listener);
    }

    @Override
    public int isFeatureSupported(String feature) {
        return billingClient.isFeatureSupported(feature);
    }

    @Override
    public boolean isReady() {
        return billingClient.isReady();
    }

    @Override
    public void startConnection(@NonNull BillingClientStateListener listener) {
        billingClient.startConnection(listener);
    }

    @Override
    public void endConnection() {
        billingClient.endConnection();
    }

    @Override
    public int launchBillingFlow(Activity activity, BillingFlowParams params) {
        return billingClient.launchBillingFlow(activity, params);
    }

    @Override
    public Purchase.PurchasesResult queryPurchases(String skuType) {
        return billingClient.queryPurchases(skuType);
    }

    @Override
    public void querySkuDetailsAsync(SkuDetailsParams params, SkuDetailsResponseListener listener) {
        billingClient.querySkuDetailsAsync(params, listener);
    }

    @Override
    public void consumeAsync(String purchaseToken, ConsumeResponseListener listener) {
        billingClient.consumeAsync(purchaseToken, listener);
    }

    @Override
    public void queryPurchaseHistoryAsync(String skuType, PurchaseHistoryResponseListener listener) {
        billingClient.queryPurchaseHistoryAsync(skuType, listener);
    }
}

File: lib/src/main/java/com/nytimes/android/external/registerlib/GoogleServiceProviderTesting.kt
package com.nytimes.android.external.registerlib

import android.app.Activity
import android.content.Context
import com.android.billingclient.api.*

class GoogleServiceProviderTesting(context: Context, listener: PurchasesUpdatedListener) : GoogleServiceProvider() {

    private val billingClient: BillingClientTesting = BillingClientTesting(context, listener)

    override fun isReady(): Boolean {
        return billingClient.isReady
    }

    override fun isFeatureSupported(feature: String): Int {
        return billingClient.isFeatureSupported(feature)
    }

    override fun startConnection(listener: BillingClientStateListener) {
        billingClient.startConnection(listener)
    }

    override fun endConnection() {
        billingClient.endConnection()
    }

    override fun launchBillingFlow(activity: Activity, params: BillingFlowParams): Int {
        return billingClient.launchBillingFlow(activity, params)
    }

    override fun queryPurchases(skuType: String): Purchase.PurchasesResult {
        return billingClient.queryPurchases(skuType)
    }

    override fun querySkuDetailsAsync(params: SkuDetailsParams, listener: SkuDetailsResponseListener) {
        billingClient.querySkuDetailsAsync(params, listener)
    }

    override fun consumeAsync(purchaseToken: String, listener: ConsumeResponseListener) {
        billingClient.consumeAsync(purchaseToken, listener)
    }

    override fun queryPurchaseHistoryAsync(skuType: String, listener: PurchaseHistoryResponseListener) {
        billingClient.queryPurchaseHistoryAsync(skuType, listener)
    }
}
